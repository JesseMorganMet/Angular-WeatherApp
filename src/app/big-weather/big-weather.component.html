<div id="bigBit">
  <h2>{{placeName}}</h2>
  <img src= "{{data.Rep | weatherType}}" alt= "{{data.Rep | weatherTypeAlt}}">
  <h2>{{data.Rep | wType}}</h2>
</div>

<!--collums-->
<div id="dataText">
<!--  rows-->
  <div>
    <h2>High's Of</h2>
    <h2>{{data.Rep | maxTemp}}&deg;C</h2>
  </div>
  <div>
    <h2>Lows Of</h2>
    <h2>{{data.Rep | minTemp}}&deg;C</h2>
  </div>
  <div>
    <h2>UV</h2>
    <h2>{{data.Rep | uv}}</h2>
  </div>
</div>
