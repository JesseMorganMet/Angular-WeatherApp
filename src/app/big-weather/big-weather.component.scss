#dataText{
  width:100%;
  display: flex;
  flex-direction: row;
  justify-content: space-evenly;
  align-items: center;
}
#dataTextIn{
  margin:0.5%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}
#bigBit{
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;

}
#bigBit > img{
  max-width: 400px;
  margin: -5% 0 -5% 0;
}
