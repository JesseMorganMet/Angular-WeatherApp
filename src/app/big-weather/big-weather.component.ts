import {Component, Input, OnInit} from '@angular/core';

@Component({
  selector: 'app-big-weather',
  templateUrl: './big-weather.component.html',
  styleUrls: ['./big-weather.component.scss']
})
export class BigWeatherComponent implements OnInit {

  @Input() placeName;
  @Input() data;

  constructor() { }

  ngOnInit(): void {
  }
}
