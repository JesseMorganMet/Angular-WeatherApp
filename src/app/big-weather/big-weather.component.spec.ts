import { ComponentFixture, TestBed } from '@angular/core/testing';

import { BigWeatherComponent } from './big-weather.component';
import {maxTempPipe, minTempPipe, uvPipe, WeatherTypeAltPipe, WeatherTypePipe, WTypePipe} from '../avg-temp.pipe';

describe('BigWeatherComponent', () => {
  let component: BigWeatherComponent;
  let fixture: ComponentFixture<BigWeatherComponent>;

  const fakeData = {
    "Rep": [{
      "T": "17",
      "W": "7",
      "U": "1"
    },
      {
        "T": "18",
        "W": "15",
        "U": "2"
      },
      {
        "T": "20",
        "W": "12",
        "U": "4"
      },
      {
        "T": "23",
        "W": "30",
        "U": "3",
      },
      {
        "T": "21",
        "W": "7",
        "U": "1"
      },
      {
        "T": "20",
        "W": "7",
        "U": "0"
      }
    ]
  };
  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [
        BigWeatherComponent,
        WeatherTypePipe,
        WeatherTypeAltPipe,
        WTypePipe,
        maxTempPipe,
        minTempPipe,
        uvPipe
      ]})
    .compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(BigWeatherComponent);
    component = fixture.componentInstance;
  });

  it('Should render the page correctly', () => {
    component.placeName = "Place"
    component.data = fakeData;
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

});

