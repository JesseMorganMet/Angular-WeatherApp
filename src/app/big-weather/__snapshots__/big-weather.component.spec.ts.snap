// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BigWeatherComponent Should render the page correctly 1`] = `
<app-big-weather
  __ngContext__={[Function LRootView]}
  data={[Function Object]}
  placeName={[Function String]}
>
  <div
    id="bigBit"
  >
    <h2>
      Place
    </h2>
    <img
      alt="Cloudy icon"
      src="../assets/icons/Cloud.png"
    />
    <h2>
      Cloudy
    </h2>
  </div><div
    id="dataText"
  >
    <div
      id="dataTextIn"
    >
      <h2>
        Highest Temperature
      </h2>
      <h2>
        23
      </h2>
    </div>
    <div
      id="dataTextIn"
    >
      <h2>
        Lowest Temperature
      </h2>
      <h2>
        17
      </h2>
    </div>
    <div
      id="dataTextIn"
    >
      <h2>
        UV
      </h2>
      <h2>
        L
      </h2>
    </div>
  </div>
</app-big-weather>
`;
